microsoft-edge/webview2/concepts/windowed-vs-visual-hosting.md

```diff
@@ -1,436 +1,185 @@
 
-If your WebView2 app uses visual hosting:
+Window to Visual hosting does not require you to use WebView2 Visual hosting APIs.
 
-* Inputs are routed to the app's `HWND` and must be configured to send the spatial input (for example, mouse, touch, and pen) based on positions, _not_ what currently has focus, such as a keyboard.
+To enable Window to Visual hosting, the environment variable `COREWEBVIEW2_FORCED_HOSTING_MODE` must be set to the value `COREWEBVIEW2_HOSTING_MODE_WINDOW_TO_VISUAL` before initializing your WebView2.
 
-The following APIs can be used when configuring WebView2 in a visual hosting environment:
+In Window-to-Visual hosting and Visual hosting, a _visual_ is a basic graphical unit that can be used to compose graphical experiences on Windows. The Windows graphics APIs that expose this functionality and are relevant to WebView2 are `DirectComposition` and `Windows.UI.Composition`. The "visual" in "Visual hosting" can be any one of `IDCompositionVisual`, `IDCompositionTarget`, or `Windows.UI.Composition.Visual`, which are visuals that are exposed through the `DirectComposition` and `Windows.UI.Composition` APIs.  (Window to Visual hosting uses `IDCompositionVisual` specifically.)  See:
+* [Basic concepts](/windows/win32/directcomp/basic-concepts) in the Windows App Development > DirectComposition docs.
+* [Composition visual](/windows/uwp/composition/composition-visual-tree) in the Windows App Development > UWP docs.
 
 
 <!-- ------------------------------ -->
-#### Composition-based rendering
+#### Advantages
 
-For composition-based WebView2 rendering, use the `CoreWebView2Environment` to create a `CoreWebView2CompositionController`. The `CoreWebView2CompositionController` implements all the APIs as `CoreWebView2Controller`, but includes additional APIs specific to composition-based rendering.
+* Different apps that share a WebView2 user data folder can have different DPI awareness.
 
-##### [.NET/C#](#tab/dotnetcsharp)
+* Different apps that share a WebView2 user data folder can have different integrity levels.
 
-* [CoreWebView2CompositionController Class](/dotnet/api/microsoft.web.webview2.core.corewebview2compositioncontroller)
-* [CoreWebView2Environment.CreateCoreWebView2CompositionControllerAsync Method](/dotnet/api/microsoft.web.webview2.core.corewebview2environment.createcorewebview2compositioncontrollerasync)
+* Different apps that share a WebView2 user data folder won't potentially cause each other to hang.
 
-##### [WinRT/C#](#tab/winrtcsharp)
+* Changing monitor scale when hosting a WebView2 in a VSTO Add-in doesn't sporadically hang the application.  See [VSTO Add-ins](/visualstudio/vsto/office-solutions-development-overview-vsto#vsto-add-ins) in _Office solutions development overview (VSTO)_.
```

Comment:
```suggestion
* Prevents sporadically hanging the app when changing monitor scale, when hosting a WebView2 in a VSTO Add-in.  See [VSTO Add-ins](/visualstudio/vsto/office-solutions-development-overview-vsto#vsto-add-ins) in _Office solutions development overview (VSTO)_.
```
alts:
* Changing monitor scale when hosting a WebView2 in a VSTO Add-in doesn't sporadically hang the application.
* Prevents sporadically hanging the app when changing monitor scale, when hosting a WebView2 in a VSTO Add-in.
* When hosting a WebView2 in a VSTO Add-in, changing monitor scale won't potentially cause the app to sporadically hang.
* Changing monitor scale won't potentially cause the app to sporadically hang, when hosting a WebView2 in a VSTO Add-in.

& consider wording in previous item: "won't potentially cause each other to hang."

Reply:
thanks for the suggestions. I picked one

Reply:
result:
When hosting a WebView2 in a VSTO Add-in, changing monitor scale won't potentially cause the app to hang.

https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/concepts/windowed-vs-visual-hosting?branch=pr-en-us-3271#advantages-1